lib/http2.c

```diff
@@ -505,6 +545,10 @@ static int push_promise(struct Curl_easy *data,
       goto fail;
     }
 
+    rv = set_transfer_url(data, &heads);
```

Comment:
`data` here is the parent handle, did you meant to use `newhandle` instead?

Reply:
oops, yes I did!